app/logical/related_tag_calculator.rb

```diff
@@ -38,9 +38,9 @@ def self.calculate_similar_from_sample(tag)
     candidates = convert_hash_to_array(counts, 100)
     similar_counts = Hash.new {|h, k| h[k] = 0}
     CurrentUser.without_safe_mode do
-      PostReadOnly.with_timeout(5_000, nil, {:tags => tag}) do
+      Post.with_timeout(5_000, nil, {:tags => tag}) do
```

Comment:
Is there a point to having the timeout, since this now barely stresses the database adapter?

Note also that elasticsearch-rails sets a default timeout of 5 seconds for its HTTP client. You can modify this for e.g. indexing purposes like this:
```rb
Elasticsearch::Model.client = Elasticsearch::Client.new(request_timeout: 30)
```

Reply:
I haven't had a chance to test these, but I don't see a reason to explicitly remove the timeout code since it should act as an extra safety net. I was also debating if it made sense to add in timeouts to the elasticsearch query, or if that would be wasted effort given that it should be very efficient.